### PR TITLE
Redux: When I am developing with application state, I want to have separation of concerns

### PR DESCRIPTION
**Pull request by larstroost**

### CODE REVIEW

File: components/time-entries/TimeEntries.jsx [hunk 1]
```diff
@@ -7,38 +8,37 @@ import NewTimeEntry from '../new-time-entry/NewTimeEntry';
...
   addTimeEntries = (newTimeEntry) => (
-    saveData(newTimeEntry).then(() => (
-      this.setState(({ timeEntries }) => ({
-        timeEntries: [
-          newTimeEntry,
-          ...timeEntries]
-      }))
-    )));
+    saveData(newTimeEntry).then(() => this.props.addNewTimeEntrySuccess(newTimeEntry)));
+    //
```
Comment (boris-arkenaar) [inline components/time-entries/TimeEntries.jsx]: Remove commented code

File: components/time-entries/TimeEntries.jsx [hunk 2]
```diff
@@ -62,4 +62,12 @@ class TimeEntries extends React.Component {
   }
 }
 
+TimeEntries.propTypes = {
+  // timeEntries: PropTypes.isArray(
```
Comment (boris-arkenaar) [inline components/time-entries/TimeEntries.jsx]: `timeEntries` is still a prop you use right?

File: components/time-entries/TimeEntries.jsx [hunk 3]
```diff
@@ -7,38 +8,37 @@ import NewTimeEntry from '../new-time-entry/NewTimeEntry';
...
 
 class TimeEntries extends React.Component {
-  state = {
-    timeEntries: []
-  }
-
   componentDidMount() {
     fetchData()
-      .then((timeEntries) => this.setState({ timeEntries }));
+      .then((timeEntries) => this.props.requestTimeEntriesSuccess(timeEntries));
```
Comment (boris-arkenaar) [inline components/time-entries/TimeEntries.jsx]: You are not using the `requestTimeEntries` and `addNewTimeEntry` actions.

File: components/time-entries/TimeEntriesContainer.jsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+import TimeEntries from './TimeEntries';
+
+const mapStateToProps = (state) => ({
+  timeEntries: timeEntriesSelector(state)
+});
+
+const mapDispatchToProps = (dispatch) => bindActionCreators({
+  requestTimeEntries,
+  requestTimeEntriesSuccess
```
Comment (boris-arkenaar) [inline components/time-entries/TimeEntriesContainer.jsx]: You are not passing along the `addNewTimeEntry` and -`Succes` actions. They are needed too right?

File: api/time-entries.json [hunk 5]
```diff
@@ -1,24 +1,29 @@
...
+    {
+      "activity": "Developing",
+      "client": "Humanoids",
+      "startTime": "2018-08-18T07:00:00.000Z",
+      "stopTime": "2018-08-18T15:00:00.000Z",
+      "id": 5
     }
   ]
-}
+}
```
Comment (boris-arkenaar) [inline api/time-entries.json]: You could delete this file...

File: components/time-entries/TimeEntries.jsx [hunk 6]
```diff
@@ -13,16 +13,13 @@ class TimeEntries extends React.Component {
...
-    saveData(newTimeEntry).then(() => this.props.addNewTimeEntrySuccess(newTimeEntry)));
-    //
-    // (
-    //   this.setState(({ timeEntries }) => ({
-    //     timeEntries: [
-    //       newTimeEntry,
-    //       ...timeEntries]
-    //   }))
-    // )));
+  onDelete = (id) => {
```
Comment (boris-arkenaar) [inline components/time-entries/TimeEntries.jsx]: An event handler should have a name starting with `handle`; `handleDelete`.

File: components/time-entry/TimeEntry.jsx [hunk 7]
```diff
@@ -1,41 +1,46 @@
...
-      </button>
-      <div>
-        <p className="single-time-entry__hours">
-          {new Date(startTime).toLocaleTimeString('nl-NL', { hour: '2-digit', minute: '2-digit' })} - {new Date(stopTime).toLocaleTimeString('nl-NL', { hour: '2-digit', minute: '2-digit' })}
-        </p>
-        <p className="single-time-entry__hours-worked">
-          {new Date((Date.parse(stopTime) - Date.parse(startTime)) - 1 * 1000 * 60 * 60).toLocaleTimeString('nl-NL', { hour: '2-digit', minute: '2-digit', seconds: '2-digit ' })}
+  client, startTime, stopTime, id, onDelete
+}) => {
+  const handleClick = () => {
```
Comment (boris-arkenaar) [inline components/time-entry/TimeEntry.jsx]: Prefer to create a class when adding functions.

File: components/time-entry/TimeEntry.jsx [hunk 8]
```diff
@@ -1,41 +1,46 @@
...
+        <button
+          className="single-time-entry__delete"
+          type="button"
+          onClick={handleClick}
+        >
+          Delete
+        </button>
+        <div>
+          <p className="single-time-entry__hours">
+            {new Date(startTime).toLocaleTimeString('nl-NL', { hour: '2-digit', minute: '2-digit' })} - {new Date(stopTime).toLocaleTimeString('nl-NL', { hour: '2-digit', minute: '2-digit' })}
```
Comment (boris-arkenaar) [inline components/time-entry/TimeEntry.jsx]: These lines are very long...

File: services/time-entries/time-entries.js [hunk 9]
```diff
@@ -1,14 +1,15 @@
...
-export const saveData = (newTimeEntry) => fetch('http://localhost:3001/api/time-entries', {
-  method: 'POST',
-  body: JSON.stringify(newTimeEntry),
-  headers: {
-    'Content-Type': 'application/json'
-  }
-})
-  .then((response) => response.json());
+export const saveData = (newTimeEntry) => (
+  fetch('http://localhost:3001/api/time-entries', {
```
Comment (boris-arkenaar) [inline services/time-entries/time-entries.js]: You are repeating `http://localhost:3001/api/time-entries`. You can put it in a constant at the top of this file.

File: ducks/time-entries.js [hunk 10]
```diff
@@ -21,6 +23,15 @@ export function timeEntriesReducer(state = initialState, action) {
       return { ...state, isLoading: true };
     case ADD_NEW_TIME_ENTRY_SUCCESS:
       return { ...state, isLoading: false, items: [action.newTimeEntry, ...state.items] };
+    case DELETE_TIME_ENTRY:
+      return { ...state, isLoading: true };
+    case DELETE_TIME_ENTRY_SUCCESS:
+      return {
+        ...state,
+        isLoading: false,
+        items: [...state.items.filter((currentEntry) => currentEntry.id
```
Comment (boris-arkenaar) [inline ducks/time-entries.js]: Je kopieert nu de volledige inhoud van een array, in een nieuwe array, zonder daar verder nog iets anders aan toe te voegen. Maw. de spread operator (en de wrapping array `[ ... ]`) is niet nodig; het resultaat van de `filter` functie is een array die je direct kan gebruiken.

File: components/time-entries/TimeEntries.jsx (hunk 3 above)
Comment (larstroost) [reply components/time-entries/TimeEntries.jsx]: True, they are already there for future use

File: services/time-entries/time-entries.js [hunk 11]
```diff
@@ -1,16 +1,19 @@
...
-export const deleteData = (timeEntryId) => fetch(`http://localhost:3001/api/time-entries/${timeEntryId}`, {
+export const saveData = (newTimeEntry) => (
+  fetch(api, {
+    method: 'POST',
+    body: JSON.stringify(newTimeEntry),
+    headers: {
+      'Content-Type': 'application/json'
+    }
+  })
+    .then((response) => response.json()));
```
Comment (tlorent) [inline services/time-entries/time-entries.js]: Perhaps place ``.then`` behind line 13, after ``})``? For readability purposes. Same applies to line 18 & 19.

File: components/time-entries/TimeEntries.jsx [hunk 12]
```diff
@@ -1,44 +1,41 @@
...
+  calculateTotalTime = (timeEntry) => this.props.timeEntries
     .filter((item) => new Date(item.startTime).toLocaleDateString()
       === new Date(timeEntry.startTime).toLocaleDateString())
     .reduce((accumulator, item) => accumulator
       + calculateDuration(item.startTime, item.stopTime), 0)
 
   render() {
+    const { timeEntries } = this.props;
     return (
       <React.Fragment>
```
Comment (tlorent) [inline components/time-entries/TimeEntries.jsx]: This ``<React.Fragment>`` is not closed? The same applies to line 36, ``<React.Fragment key={timeEntry.id}>``. [https://reactjs.org/docs/fragments.html](url)

**Pull request merged**